### PR TITLE
#166789954 Users should be able to Login via Google Facebook or Twitter

### PR DESCRIPTION
**Pull request by elemanhillary-zz**
#### What does this PR do?
- adds social login buttons
#### Description of Task to be completed
- enables social login functionality
#### How should this be manually tested?
```
git clone https://github.com/andela/ah-92explorers-frontend.git
cd ah-92explorers-frontend
git ft-login-via-social-sites-166789954
configure .env file `BASE_URL` value
npm install
npm run dev
```
- Navigate to `http://localhost:8080/login`
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
[#166789954](https://www.pivotaltracker.com/story/show/166789954)
#### Screenshots (if appropriate)
Imgur
#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by Hervera — CHANGES_REQUESTED** [review]
Fix the circleCI build error

**Review by IsaiahRn — CHANGES_REQUESTED** [review]
From the screenshot that you have provided, the twitter icon is missing.
Kindly, add that icon?

File: src/__tests__/__actions__/socialLogin.spec.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  // });
+
+  // afterEach(() => {
+  //   windowSpy.mockRestore();
+  // });
+  it('should return LOGIN_SUCCESS action with user info', () => {
+    // global.window = Object.create(window);
+    const url = '?token=0f362c370ad9d1d4546c1ac6c8e36e94dfa0ef81ce71e1dabc22753b86d30b6feb7b6432773ebbd5cdcc329a0ef523bce5050aea4d79bb809c56683aae2ab609971b994d920499e1b2b25afc98d2c4f1a2451ca7931695eec685aeede7c37b2302bdd4cebcbbe35a04eaab8cccc73ef015109976fda2b99263d64f35fa79e1adba449772e5901994a9913ba071dc229a06a96f6f2a118e148d7a02139f6019198c11319ab32043d417e767258d677a8a005473835be1c33a0cccd8f2f4251ebe5245cdbcfb4a07d620e51872ef830e5d339417e3d8e95a6cbbc626af65046947b211ababeac4ba082f10c98f252120aa8a484f225e2914c37e73ec325f59489bafd341b0d087600ed63ad7cd464eaabd37d9a5d3750d6db143449a4c0ea178b4e469ff2aa1946768d649f33d9cb63c102ad0607f3bcf99c531d067b84ec2b1f9e8e4072e59a98abf9dea719511daea2238f89d1f3b531923dc54dc3c0d5b18b74f3d884554f76b8b4d4696fd4120c7c10a09b640ca69';
+    // windowSpy.mockImplementation(() => ({
+    //   location: {
```
Comment (IsaiahRn) [inline src/__tests__/__actions__/socialLogin.spec.js]: Can you remove all these commented lines of code?

**Review by Hervera — CHANGES_REQUESTED** [review]
Remove conflicts

File: src/__tests__/components/social.spec.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const middlewares = [thunk, promiseMiddleware];
+const mockStore = configureMockStore(middlewares);
+const store = mockStore({});
+describe('Render SocialLogin component', () => {
+  it('to have wrapper class', async () => {
+    const wrapper = mount(<Provider store={store}><Social /></Provider>);
+    const btn = wrapper.find('.twitter').first();
+    btn.simulate('click');
+    expect(btn.length).toBe(1);
```
Comment (kodek-sleuth) [inline src/__tests__/components/social.spec.js:18]: I have realised that you not actually testing what the button will do after stimulation. This is universal for All the three buttons, Please kindly indicate those tests there

File: src/components/continue/continue.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React, { Component, Fragment } from 'react';
```
Comment (kodek-sleuth) [inline src/components/continue/continue.js]: The folder name should start with an Uppercase letter and all files with `jsx` should have the `jsx` extension. This is based on The Team Coding Conventions we signed up for

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

File: src/components/social/social.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import dotenv from 'dotenv';
+import { social } from '../../redux/actions/actionCreators';
+
+dotenv.config();
+
+const { BASE_URL } = process.env;
+class Social extends Component {
+  handleOnClick = (e) => {
+    const { social: redirect } = this.props;
+    const classNames = e.target.className.split(' ');
```
Comment (elemanhillary-zz) [inline src/components/social/social.js]: use a unique` id` for each of the buttons.

File: src/components/social/social.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { Button } from 'reactstrap';
+import dotenv from 'dotenv';
+import { social } from '../../redux/actions/actionCreators';
+
+dotenv.config();
+
+const { BASE_URL } = process.env;
+class Social extends Component {
+  handleOnClick = (e) => {
+    const { social: redirect } = this.props;
```
Comment (elemanhillary-zz) [inline src/components/social/social.js]: rename to `redirectURL`

File: src/redux/actions/actionCreators/social.action.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export const getInfo = (search) => (dispatch) => {
+  const stringObj = qs.parse(search);
+  const tokenStr = stringObj && stringObj.token;
+  if (!tokenStr) {
+    const error = 'something went wrong';
+    return dispatch(failure(error));
+  }
+  try {
+    const data = dec(tokenStr);
```
Comment (elemanhillary-zz) [inline src/redux/actions/actionCreators/social.action.js:37]: use `async/await` here

**Review by niyoceles — APPROVED** [review]

**Review by IsaiahRn — APPROVED** [review]
It can be merged

**Review by Hervera — APPROVED** [review]
Ready to be merged

**Review by kodek-sleuth — APPROVED** [review]

**Pull request merged**